app/lifecycle/updater.go

```diff
@@ -89,28 +89,33 @@ func DownloadNewRelease(updateResp UpdateResponse) error {
 		if n, err := fp.Write(payload); err != nil || n != len(payload) {
 			return fmt.Errorf("write payload %s: %d vs %d -- %w", escapedFilename, n, len(payload), err)
 		}
-		log.Printf("XXX completed writing out update payload to %s", escapedFilename)
+		slog.Debug(fmt.Sprintf("XXX completed writing out update payload to %s", escapedFilename))
 	} else if err != nil {
 		return fmt.Errorf("XXX unexpected stat error %w", err)
 	} else {
-		log.Printf("XXX update already downloaded")
+		slog.Debug("XXX update already downloaded")
 	}
 	UpdateDownloaded = true
 	return nil
 }
 
-func StartBackgroundUpdaterChecker(ctx context.Context, cb func(bool)) {
-	// TODO - shutdown mechanism
+func StartBackgroundUpdaterChecker(ctx context.Context, cb func(string) error) {
```

Comment:
Given this spawns a goroutine we might just want to let the caller decide concurrency and remove the `go func () {...`

Reply:
I'll pick this up on my next pass of refinements.